### PR TITLE
Feature/TimeRanges

### PR DESCRIPTION
**Pull request by lukas-zech-software**
Datepicker now uses time ranges to determine the start and end date of the current displayed period

To support big time ranges, dates now render via lazy loading

http://jira.media-control.int/browse/PM-18?filter=-1
please review: @gfke/frontend
Also see https://github.com/gfke/planet-music/pull/16

### CODE REVIEW

File: source/yosoDatepickerDirective.spec.js [hunk 1]
```diff
@@ -71,8 +72,29 @@ describe('yosoDatePickerDirective', function () {
             availablePeriodModes: [period.PERIOD_MODES.DAYS, period.PERIOD_MODES.WEEKS, period.PERIOD_MODES.MONTHS],
             defaultPeriodMode: period.PERIOD_MODES.DAYS,
             maxSelectablePeriods: 3,
-            startDate: new Date(),
-            endDate: new Date(),
+            timeRanges: {
+                'd': {
+                    'bwg': {
+                        'min': '2005-01-02T23:00:00.000Z',
```
Comment (mstein77) [inline source/yosoDatepickerDirective.spec.js]: Does the timeRange start and end value really need a time and locale information, or is it also possible to use a YYYY-MM-DD utc-string?

Comment (lukas-zech-software) [inline source/yosoDatepickerDirective.spec.js]: I just copied that from a real JSON response

There is no need to define it in that format.
I'll remove it for simplicity's sake

Mit freundlichen Grüßen,

Lukas Zech
Softwareentwicklung
Kappeneck 20b
86152 Augsburg
lukas.zech.software@gmail.com
XING <https://www.xing.com/profile/Lukas_Zech3>

**Comment by mstein77** [comment]
Ready2merge

**Comment by lukas-zech-software** [comment]
merged by squash

**Pull request closed without merging**